### PR TITLE
chore(changelog): fixup git-cliff formatting

### PR DESCRIPTION
**Pull request by joshka**
- Replaced backticks surrounding body with blockquotes. This formats
  significantly better.
- Added a few postprocessors to remove junk like the PR template,
  horizontal lines and some trailing whitespace

---

Note: there is extra non-automatically generated stuff in the changelog that this would remove - the changes to CHANGELOG.md should not be merged as-is, and this is worth waiting for @orhun to check out.

Compare: https://github.com/ratatui-org/ratatui/blob/5e7fbe8c328050a282a49b8c6ab3f5b33c7cfb4c/CHANGELOG.md
To: https://github.com/ratatui-org/ratatui/blob/main/CHANGELOG.md

### CODE REVIEW

**Comment by joshka** [comment]
> It seems to be missing a summary section

Yep - I mentioned this in the PR comment and the reason for the do not merge label.

> Note: there is extra non-automatically generated stuff in the changelog that this would remove - the changes to CHANGELOG.md should not be merged as-is, and this is worth waiting for @orhun to check out.

File: CHANGELOG.md
Comment (orhun) [inline CHANGELOG.md:1]: We can revert the changes in CHANGELOG.md and start using this new format for the upcoming releases. There are already old formats in the changelog since we manually append it each time anyways.

File: cliff.toml [hunk 1]
```diff
@@ -87,6 +85,12 @@ trim = false
 footer = """
 <!-- generated by git-cliff -->
 """
+postprocessors = [
+  # { pattern = '<REPO>', replace = "https://github.com/joshka/focusable" },                                # replace repository URL
```
Comment (orhun) [inline cliff.toml:89]: not needed?
```suggestion
```

**Review by orhun — APPROVED** [review]

File: cliff.toml (hunk 1 above)
Comment (joshka) [reply cliff.toml:89]: woops - bad copy paste

File: CHANGELOG.md
Comment (joshka) [reply CHANGELOG.md:1]: Makes sense, though the manual part of this feeds into the issue about releasing more frequently and less manually.

Comment (orhun) [reply CHANGELOG.md:1]: Yup, commented in #1232 about this.

Also, we can use the following command for appending the unreleased to the changelog without changing the previous entries:

```sh
git cliff -l -p CHANGELOG.md
```

**Review by orhun — APPROVED** [review]

**Comment by joshka** [comment]
Thanks! @orhun
